integration/query_frontend_test.go

```diff
@@ -455,6 +455,12 @@ func runQueryFrontendTest(t *testing.T, cfg queryFrontendTestConfig) {
 	assertServiceMetricsPrefixes(t, Querier, querier)
 	assertServiceMetricsPrefixes(t, QueryFrontend, queryFrontend)
 	assertServiceMetricsPrefixes(t, QueryScheduler, queryScheduler)
+
+	// Ensure cortex_distributor_replication_factor is not exported when ingest storage is enabled
+	// because it's how we detect whether a Mimir cluster is running with ingest storage.
+	if flags["-ingest-storage.enabled"] == "true" {
+		assertServiceMetricsNotMatching(t, "cortex_distributor_replication_factor", queryFrontend, queryScheduler, distributor, ingester, querier)
+	}
```

Comment:
Do we have any tests that verify this metric presents otherwise?

Reply:
Good point, we don't. Added in https://github.com/grafana/mimir/pull/9385/commits/0062f2d7cb7cb1f0db7a113575fc0ed0edd9b18c.